### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by benlapETS**
Completed the alarm clock #2. What I did:
- Set up my workspace using pycharm.
- Update the project README to document the essential for each project.
- Also completed the documentation for the actual project: the Alarm.
- Completed the code for the alarm (model).
- Completed the code for the tests (alarm_test).
- Completed the code for the MVC architecture (view and controller).
- Also documented each module properly which would establish the standard for the remaining projects.

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<component name="InspectionProjectProfileManager">
+  <profile version="1.0">
+    <option name="myName" value="Project Default" />
+    <inspection_tool class="PyUnresolvedReferencesInspection" enabled="true" level="WARNING" enabled_by_default="true">
+      <option name="ignoredIdentifiers">
+        <list>
+          <option value="sarra.sr_cache.check.mock_calls" />
```
Comment (sriendea) [inline .idea/inspectionProfiles/Project_Default.xml:7]: value="sarra.sr_cache.check.mock_calls"

I am not sure your suppose to have that kind of config here.

File: README.md [hunk 2]
```diff
@@ -1,12 +1,41 @@
...
-- [ ] Hangman Game in Python
-- [ ] Dice Rolling Simulator Python Game
-- [ ] Password Generator
-- [ ] Currency Converter
-- [ ] Text to Speech
-- [ ] Message Encode Decode in Python with Tkinter
-- [ ] COVID-19 Spread Analysis with Flask
-- [ ] Simple Calculator
-- [ ] Library Managememnt System
+## Alarm Clock with GUI
```
Comment (sriendea) [inline README.md:3]: Maybe, you can think about adding the link of the project in the Read Me file

File: alarm/__init__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        The application of an alarm
+    **alarm_test.py**:
+        The tests of an alarm
+    **controller.py**:
+        The controller of an alarm
+    **model.py**:
+        The model of an alarm
+    **view.py**:
+        The GUI of an alarm
+"""
```
Comment (sriendea) [inline alarm/__init__.py:22]: Add a new line at the end of the file.

File: alarm/model.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+FFPLAY_CMD = ['ffplay', '-nodisp', '-autoexit', '-hide_banner', '-v', '0', '/usr/share/sounds/purple/login.wav']
+
+# Logging messages
+MSG_INFO_START = "Starting alarm"
+MSG_INFO_STOP = "Stopping alarm"
+MSG_DEBUG_SLEEP = "self.alarm_time={}, self.sleep_time={}"
+MSG_INFO_WAKEUP = "Wake up Dude"
+
+# Time related
+ATIME = namedtuple('at', ['hour', 'min', 'sec'])
```
Comment (sriendea) [inline alarm/model.py:87]: Does 'at' is in fact 'aTime', to be more consistent?

File: alarm/view.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+from tkinter.font import Font
+
+# View constants
+ENTRY_ROW = 1
+ENTRY_WIDTH = 2
+LABEL_ENTRY_ROW = 0
+MIN_WIDTH = 120
+MIN_HEIGHT = 250
+PAD_X = 10
+TITLE = "Labesoft Alarm"
```
Comment (sriendea) [inline alarm/view.py:76]: I prefer " LABESOFT Alarm " as title

File: alarm/model.py (hunk 4 above)
Comment (benlapETS) [reply alarm/model.py:87]: I put ATIME in all caps as it is a constant.

File: alarm/view.py (hunk 5 above)
Comment (benlapETS) [reply alarm/view.py]: You already resolved it thx

**Pull request merged**